python/equistore-core/tests/tensor.py

```diff
@@ -361,3 +363,35 @@ def test_components_to_properties(tensor):
     assert tuple(block.properties[0]) == (0, 0)
     assert tuple(block.properties[1]) == (1, 0)
     assert tuple(block.properties[2]) == (2, 0)
+
+
+def test_empty_tensor():
+    empty_tensor = TensorMap(keys=Labels.empty(["key"]), blocks=[])
+
+    assert empty_tensor.keys.names == ("key",)
+
+    assert empty_tensor.sample_names == tuple()
+    assert empty_tensor.components_names == []
+    assert empty_tensor.property_names == tuple()
+
+    assert empty_tensor.blocks() == []
+
+    assert empty_tensor.blocks_matching(key=3) == []
+
+    message = "invalid parameter: 'not_a_key' is not part of the keys for this tensor"
+    with pytest.raises(EquistoreError, match=message):
+        empty_tensor.blocks_matching(not_a_key=3)
+
+    message = (
+        "invalid parameter: block index out of bounds: we have "
+        "0 blocks but the index is 3"
+    )
+    with pytest.raises(EquistoreError, match=message):
+        empty_tensor.block(3)
```

Comment:
Would be good to have a test on `blocks()` that also raises an error

Reply:
done!